src/runtime/particle-spec.ts

```diff
@@ -245,10 +251,18 @@ export class ParticleSpec {
     return new ParticleSpec({args, name, verbs: verbs || [], description, implFile, modality, slots});
   }
 
-  clone() {
+  clone(): ParticleSpec {
     return ParticleSpec.fromLiteral(this.toLiteral());
   }
 
+  cloneWithResolutions(variableMap: Map<TypeVariableInfo|Schema, TypeVariableInfo|Schema>): ParticleSpec {
```

Comment:
Can you add a comment here (and on clone() above, I think) that states these shouldn't be called directly?

Reply:
added